fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/storage/Storage.java

```diff
@@ -247,4 +212,48 @@ default long getVersion() {
 	static <T> Class<Storage<T>> asClass() {
 		return (Class<Storage<T>>) (Object) Storage.class;
 	}
+
+	/**
+	 * Convenient helper to simulate an insertion, i.e. get the result of insert without modifying any state.
+	 * The passed transaction may be null if a new transaction should be opened for the simulation.
+	 * @see #insert
+	 * @deprecated Either use transactions directly, or use {@link StorageUtil#simulateInsert}.
+	 */
+	@Deprecated(forRemoval = true)
+	default long simulateInsert(T resource, long maxAmount, @Nullable TransactionContext transaction) {
+		try (Transaction simulateTransaction = Transaction.openNested(transaction)) {
```

Comment:
The implementation should be switched to call the StorageUtil methods directly, so users know what they should call instead even if the javadoc is not available and reduces changes if the simulate implementation needs an update.